From 1b93f45fbd144cdf7095102bf6927c97019cf2e7 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Tue, 20 Oct 2015 09:03:25 +0200
Subject: [PATCH] ext master secret: extension is marked as mandatory

This forces the extension to be sent even where resuming sessions.
Resolves #45
---
 lib/ext/ext_master_secret.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ext/ext_master_secret.c b/lib/ext/ext_master_secret.c
index 9610f12..0d90347 100644
--- a/lib/ext/ext_master_secret.c
+++ b/lib/ext/ext_master_secret.c
@@ -38,7 +38,7 @@ static int _gnutls_ext_master_secret_send_params(gnutls_session_t session,
 extension_entry_st ext_mod_ext_master_secret = {
 	.name = "EXT MASTER SECRET",
 	.type = GNUTLS_EXTENSION_EXT_MASTER_SECRET,
-	.parse_type = GNUTLS_EXT_TLS,
+	.parse_type = GNUTLS_EXT_MANDATORY,
 
 	.recv_func = _gnutls_ext_master_secret_recv_params,
 	.send_func = _gnutls_ext_master_secret_send_params,
-- 
2.5.0

